src/views/catalog/templatescatalog/hooks/useAvailableDataSourcesAndPVCs.ts

```diff
@@ -1,19 +1,22 @@
 import React from 'react';
 
-import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { PersistentVolumeClaimModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import DataSourceModel from '@kubevirt-ui/kubevirt-api/console/models/DataSourceModel';
 import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
 import { BOOT_SOURCE } from '@kubevirt-utils/resources/template';
 import {
-  getDataSource,
-  getPVC,
   getTemplateBootSourceType,
   isDataSourceReady,
 } from '@kubevirt-utils/resources/template/hooks/useVmTemplateSource/utils';
-
-import { assertFulfilled, getSourcePromises } from './utils';
+import {
+  getGroupVersionKindForModel,
+  useK8sWatchResources,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
 
 type UniqueSourceType = {
-  [namespace: string]: Set<string>;
+  [key in string]: WatchK8sResource;
```

Comment:
why did you changed to `key in string`?

Reply:
because now the object key is not the namespace but is `NAMESPACE-NAME`